app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+# frozen_string_literal: true
+# Allows for the creation of multiple tube racks and their tubes.
+# rubocop:disable Metrics/ClassLength
+class SpecificTubeRackCreation < AssetCreation
+  # Allows a many to many relationship between SpecificTubeRackCreations and child Tube racks.
+  class ChildTubeRack < ApplicationRecord
+    self.table_name = ('specific_tube_rack_creation_children')
+    belongs_to :specific_tube_rack_creation
+    belongs_to :tube_rack
+  end
+
+  # Allows a many to many relationship between SpecificTubeRackCreations and Purposes.
+  class ChildPurpose < ApplicationRecord
+    self.table_name = 'specific_tube_rack_creation_purposes'
+    belongs_to :specific_tube_rack_creation
+    belongs_to :tube_rack_purpose, class_name: 'Purpose'
+  end
+
+  has_many :creation_child_purposes, class_name: 'SpecificTubeRackCreation::ChildPurpose', dependent: :destroy
+  has_many :child_purposes, through: :creation_child_purposes, source: :tube_rack_purpose
+
+  has_many :child_tube_racks, class_name: 'SpecificTubeRackCreation::ChildTubeRack', dependent: :destroy
+  has_many :children, through: :child_tube_racks, source: :tube_rack
+
+  validates :tube_rack_attributes, presence: true
+
+  has_many :parent_associations,
+           foreign_key: 'asset_creation_id',
+           class_name: 'AssetCreation::ParentAssociation',
+           inverse_of: 'asset_creation',
+           dependent: :destroy
+
+  # NB. assumption that parent is a plate
+  belongs_to :parent, class_name: 'Plate'
+  has_many :parents, through: :parent_associations, class_name: 'Plate'
+
+  #   @param value [Array<Hash>] Hashes defining the attributes to apply to each tube rack and
+  #     the tubes within that are being created.
+  #     This is used to set custom attributes on the tube racks, such as name. As well as to create
+  #     the tubes within the tube rack and link them together.
+  #   @example [
+  #   {
+  #     :tube_rack_name=>"Tube Rack",
+  #     :tube_rack_barcode=>"TR00000001",
+  #     :tube_rack_purpose_uuid=>"0ab4c9cc-4dad-11ef-8ca3-82c61098d1a1",
+  #     :tube_rack_metadata_key=>"tube_rack_barcode",
+  #     :racked_tubes=>[
+  #       {
+  #         :tube_barcode=>"SQ45303801",
+  #         :tube_name=>"SEQ:NT749R:A1",
+  #         :tube_purpose_uuid=>"0ab4c9cc-4dad-11ef-8ca3-82c61098d1a1",
+  #         :tube_position=>"A1",
+  #         :parent_uuids=>["bd49e7f8-80a1-11ef-bab6-82c61098d1a0"]
```

Comment:
What is `parent_uuids`? Should this be `rack_uuid`? But the rack link is already specified. Is it actual parent labware? I thought the parent-child relationships were going on the racks?

Reply:
These are the parent plate well uuids for this specific tube. Required to set up the transfer requests.
racked_tubes is nested within a specific tube rack definition which we've supplied the tube rack purpose uuid for. Remember we haven't created the rack or tubes yet so we don't have their uuids.